FlowCryptUI/Nodes/PassPhraseAlertNode.swift

```diff
@@ -0,0 +1,292 @@
+        let node = ASTextNode()
+        let font = isBold ? UIFont.boldSystemFont(ofSize: fontSize) : UIFont.systemFont(ofSize: fontSize)
+        node.attributedText = NSAttributedString(
+            string: text,
+            attributes: [NSAttributedString.Key.font: font]
+        )
+        node.accessibilityIdentifier = identifier
+        return node
+    }
+
+    private func createButtonNode(title: String, color: UIColor, identifier: String, action: Selector) -> ASButtonNode {
+        let node = ASButtonNode()
+        node.setTitle(title, with: UIFont.systemFont(ofSize: 15), with: color, for: .normal)
+        node.style.flexGrow = 1
+        node.style.preferredSize.height = 35
+        node.addTarget(self, action: action, forControlEvents: .touchUpInside)
+        node.accessibilityIdentifier = identifier
+        node.setBackgroundColor(.lightGray, forState: .highlighted)
+        return node
+    }
+
+    private func updateIntroduction() {
+        if let introduction, !introduction.isEmpty {
+            introductionLabel.attributedText = NSAttributedString(string: introduction)
+            introductionLabel.isHidden = false
+        } else {
+            introductionLabel.isHidden = true
+        }
+        setNeedsLayout()
+        contentView.setNeedsLayout()
+    }
+
+    private func createPassPhraseTextField() -> TextFieldNode {
+        let node = TextFieldNode(accessibilityIdentifier: "aid-message-passphrase-textfield") { [weak self] action in
+            if case let .didPaste(_, value) = action {
+                self?.submitPassphrase(text: value)
+            }
+        }
+        node.shouldReturn = { textField in
+            self.submitPassphrase(text: textField.text)
+            return true
+        }
+        node.borderColor = .init(gray: 0.2, alpha: 1.0)
+        node.borderWidth = 0.1
+        node.backgroundColor = .white
+        node.cornerRadius = 6
+        node.style.width = ASDimension(unit: .fraction, value: 1.0)
+        node.style.preferredSize.height = 35
+        node.textInsets = 5
+        node.isSecureTextEntry = true
+        return node
+    }
+
+    // MARK: - Layout
+    override public func layoutSpecThatFits(_ constrainedSize: ASSizeRange) -> ASLayoutSpec {
+        let separatorInsetSpec = ASInsetLayoutSpec(
+            insets: UIEdgeInsets(top: 10, left: 0, bottom: 0, right: 0),
+            child: separatorNode
+        )
+
+        let buttonStack = ASStackLayoutSpec(
+            direction: .horizontal,
+            spacing: 0,
+            justifyContent: .spaceBetween,
+            alignItems: .center,
+            children: [cancelButton, okayButton]
+        )
+        buttonStack.style.flexGrow = 1.0
+
+        var contentChildren: [ASLayoutElement] = [passPhraseTextField]
+        if message != nil {
+            contentChildren.insert(messageLabel, at: 1)
```

Comment:
can be just `contentChildren.append(messageLabel)`, as it will always be at index 1

Reply:
Sounds good!